src/Symfony/Component/ImportMap/ImportMapManager.php

```diff
@@ -0,0 +1,321 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ImportMap;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\VarExporter\VarExporter;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class ImportMapManager
+{
+    public const PROVIDER_JSPM = 'jspm';
+    public const PROVIDER_JSPM_SYSTEM = 'jspm.system';
+    public const PROVIDER_SKYPACK = 'skypack';
+    public const PROVIDER_JSDELIVR = 'jsdelivr';
+    public const PROVIDER_UNPKG = 'unpkg';
+    public const PROVIDERS = [
+        self::PROVIDER_JSPM,
+        self::PROVIDER_JSPM_SYSTEM,
+        self::PROVIDER_SKYPACK,
+        self::PROVIDER_JSDELIVR,
+        self::PROVIDER_UNPKG,
+    ];
+
+    public const POLYFILL_URL = 'https://ga.jspm.io/npm:es-module-shims@1.7.0/dist/es-module-shims.js';
+
+    /**
+     * @see https://regex101.com/r/2cR9Rh/1
+     *
+     * Partially based on https://github.com/dword-design/package-name-regex
+     */
+    private const PACKAGE_PATTERN = '/^(?:https?:\/\/[\w\.-]+\/)?(?:(?<registry>\w+):)?(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*)(?:@(?<version>[\w\._-]+))?(?:(?<subpath>\/.*))?$/';
+
+    private array $importMap;
+    private array $modulesToPreload;
+    private string $json;
+
+    public function __construct(
+        private readonly string $path,
+        private readonly string $assetsDir = 'assets/',
+        private readonly string $publicAssetsDir = 'public/assets/',
```

Comment:
if we follow Ryan's suggestion to let assets:build put files under public, is this still needed?

Reply:
I need to take a deeper look to this, but probably not.